histomicstk/features/ComputeHaralickFeatures.py

```diff
@@ -0,0 +1,210 @@
+    if len(im_intensity.shape) != 2:  # color image
+        raise ValueError("Inputs 'I' should be a grayscale image")
+
+    for i in range(numLabels):
+        # get bounds of an intensity image
+        minr, minc, maxr, maxc = rprops[i].bbox
+        # grab nucleus mask
+        subImage = im_intensity[minr:maxr+1, minc:maxc+1].astype(np.uint8)
+        # gets GLCM or gray-tone spatial dependence matrix
+        arrayGLCM = graycomatrixext(subImage, offsets=offsets,
+                                    num_levels=num_levels,
+                                    gray_limits=gray_limits)
+        num_dims = len(subImage.shape)
+        # offsets
+        if offsets is None:
+            # set default offset value
+            offsets = np.identity(num_dims)
+        else:
+            # check sanity
+            if offsets.shape[1] != num_dims:
+                raise ValueError(
+                    'Dimension mismatch between input image and offsets'
+                )
+        num_offsets = offsets.shape[0]
+        f = np.zeros((numFeatures/2, num_offsets))
+        for r in range(num_offsets):
+            # normalize GLCM
+            R = np.sum(arrayGLCM[:, :, r], dtype=np.float)
+            nGLCM = arrayGLCM[:, :, r]/R
+            # get marginal-probability matrix summing the rows
+            px = np.sum(nGLCM, axis=1)
+            py = np.sum(nGLCM, axis=0)
+            # initialize marginal-probability matrix sets
+            # summing normalizedGLCM such that i+j = k or i-j = k
+            pxPlusy = np.zeros(2*num_levels-1)
+            pxMinusy = np.zeros(num_levels)
+            # arbitarily small positive constant to avoid log 0
+            e = 0.00001
+            for n in range(0, num_levels):
+                for m in range(0, num_levels):
+                    # gets marginal-probability matrix
+                    pxPlusy[n+m] = pxPlusy[n+m] + nGLCM[n, m]
+                    pxMinusy[abs(n-m)] = pxMinusy[abs(n-m)] + nGLCM[n, m]
+            # f0: computes angular second moment
+            f[0, r] = np.sum(np.square(nGLCM))
+            # f1: computes contrast
+            n_Minus = np.arange(num_levels)
+            f[1, r] = np.dot(np.square(n_Minus), pxMinusy)
+            # f2: computes correlation
+            # gets weighted mean and standard deviation of px and py
+            meanx = np.dot(n_Minus, px)
+            variance = np.dot(px, np.square(n_Minus)) - np.square(meanx)
+            nGLCMr = np.ravel(nGLCM)
+            x, y = np.mgrid[0:num_levels, 0:num_levels]
+            xy = x*y
+            f[2, r] = (np.dot(np.ravel(xy), nGLCMr) - np.square(meanx)) / \
+                variance
+            # f3: computes sum of squares : variance
+            f[3, r] = variance
+            # f4: computes inverse difference moment
+            xy_IDM = 1. / (1+np.square(x-y))
+            f[4, r] = np.dot(np.ravel(xy_IDM), nGLCMr)
+            # f5: computes sum average
```

Comment:
insert an empty line above each of the comment like [this](https://github.com/DigitalSlideArchive/HistomicsTK/blob/master/histomicstk/features/ComputeMorphometryFeatures.py#L115). It will make the code look less cluttered and easier to read.